airqo-platform/AirQo-api
# Join analytics

- What does this PR do?
This PR created new endpoints for the join analytics use case.
- What JIRA tasks are related to this PR?
 [1. Request Access](https://airqoteam.atlassian.net/browse/NLYTCS-1).
[2. Admin set up user](https://airqoteam.atlassian.net/browse/NLYTCS-27).
[3. Log out](https://airqoteam.atlassian.net/browse/NLYTCS-28)
- How do I test out this PR?
Git clone this repo `cd AirQo-api/src/auth-service` Then run `npm install` to get all the necessary dependencies Afterwards run `npm run test` to test out the end points using any REST client tool available to you.
- Which end points are ready for testing?
[AirQo REST API documentation](https://docs.google.com/spreadsheets/d/1-NGTDBtTcIi8Fg2UYSWjcK6geKrZdopz5kxRFsCEMPs/edit?usp=sharing)
1. **Collaborator Login** > [PUT] partial url: `/api/v1/users/loginCollaborator`
2. **Delete User** > [DELETE] partial url: `/api/v1/users/_USER-ID_`
3. **Confirm Email** > [POST] partial url: `/api/v1/users/email/confirm/_USER-ID_`
4. **Register Collaborator** > [POST] partial url: `/api/v1/users/register/collab`
5. **Update Collaborator** > [PUT] partial url: `/api/v1/users/update/collab/_USER-ID_`
6. **Delete Collaborator** > [DELETE] partial url: `/api/v1/users/update/collab/_USER-ID_`
7. **Collaborator Login** > [POST] partial url: `/api/v1/users/loginCollaborator`
8. **updatePasswordViaEmail** > [PUT] partial url: `/api/v1/users/updatePasswordViaEmail`
9. **reset password** > [GET] >partial url: `/api/v1/users/reset?_TOKEN_`
10. **forgotPassword** > [POST] > partial url: `/api/v1/users/forgotPassword/_EMAIL_`
